File: 0718-maximum-length-of-repeated-subarray/solution.java
class Solution {
    public int findLength(int[] nums1, int[] nums2) {
        // this code is same like longest common subsequence of string 
        int n1 = nums1.length, n2= nums2.length;

        int [][]dp = new int[n1+1][n2+1];
        int maxLen = 0;
        for(int i=1; i<=n1; i++){
            for(int j=1; j<=n2; j++){
                if(nums1[i-1] == nums2[j-1]){
                    dp[i][j] = 1 + dp[i-1][j-1];
                }else dp[i][j] = 0;  //  no need of this line b/c 
                // in java all array elements are initialize with 0 only

                maxLen = Math.max(maxLen, dp[i][j]);
            }
        }
        return maxLen;
    }
}

File: 0718-maximum-length-of-repeated-subarray/solution.cpp
class Solution {
public:
    int findLength(vector<int>& nums1, vector<int>& nums2) {
        int n= nums1.size(), m = nums2.size();
        int maxLen = 0;
        for(int i=0; i<n; i++){
            for(int j=0; j<m; j++){
                if(nums1[i] == nums2[j]){
                    int cnt = 1;
                    int ii=i+1, jj= j+1;
                    while(ii<n && jj<m && nums1[ii] == nums2[jj]){
                        cnt++;
                        ii++; jj++;
                    }
                    maxLen = max(maxLen, cnt);
                }
            }
        }
        return maxLen;
    }
};
